### PR TITLE
build: add major, minor and patch outputs.

### PR DESCRIPTION
**Pull request by mrdoodles**
# Description

Added outputs for major_version, minor_version and patch_version to use in workflows i.e. tagging image, code, action versions available at multiple non breaking versions: v1, v1.2, v1.2.3. README updated to reflect the changes in the action.sh.

### CODE REVIEW

**Comment by TomerFi** [comment]
@mrdoodles 
This is a great idea, thank you very much.
Maybe we should also add a `minor_next_dev`,
for the next development iteration part of the version:

| Key                           | Description                                | Example     |
| --------------------------- | ----------------------------------------- | --------------- |
| `new_version`           | The next semantic version        | `1.2.3`        |
| `next_dev_iteration` | The next development iteration | `1.2.4.dev` |
| `major_version`        | The next major version              | `1`              |
| `minor_version`        | The next minor version              | `2`              |
| `patch_version`        | The next patch version              | `3`               |
| `minor_next_dev`     | The next minor development    | `4.dev`         |

**Comment by mrdoodles** [comment]
@TomerFi I forgot to add the action.yml but have subsequently added. 
I am not familiar with the .dev suffix but I presume it is similar to SNAPSHOT in the maven world. 
would you like me to add this, perhaps something as simple as:
echo "::set-output name=patch_dev_iteration::$patch_version.dev"

**Comment by mrdoodles** [comment]
@TomerFi Have added the patch_next_dev

**Comment by TomerFi** [comment]
@all-contributors add @mrdoodles for code and docs

File: action.sh [hunk 1]
```diff
@@ -15,6 +15,15 @@
 output=$(/usr/local/scripts/entrypoint.sh --changelog $1 --preset $2 --label $3 --repopath $4)
 # split output members from the variable
 read new_version next_dev_iteration <<<$(cut -f1,2 -d" " <<<$output)
+# split out major, minor, patch versions
+read major_version <<<$(cut -f1 -d"." <<<"$new_version")
+read minor_version <<<$(cut -f2 -d"." <<<"$new_version")
+read patch_version <<<$(cut -f3 -d"." <<<"$new_version")
+patch_next_dev="$((patch_version+1)).dev"
```
Comment (TomerFi) [inline action.sh:22]: The `.dev` prefix is interchangable, this should do the trick I think:

```shell
read patch_next_dev <<<$(cut -f3- -d"." <<<"$next_dev_iteration")
```

**Review by TomerFi — CHANGES_REQUESTED** [review]

File: action.sh (hunk 1 above)
Comment (TomerFi) [reply action.sh:22]: Actually, I'm already here, I can do it myself.
:-)

**Review by TomerFi — APPROVED** [review]

**Review by TomerFi — APPROVED** [review]

**Pull request merged**